### PR TITLE
feat: added ethers@5 and token-registry@2.1 and fix types

### PR DESCRIPTION
**Pull request by justussoh**
This PR upgrades the current packages of ethers and token-registry to the latest version.

Found a related bug:
https://github.com/TradeTrust/tradetrust-website/issues/241

### CODE REVIEW

File: src/common/contexts/TokenInformationContext/TokenInformationContext.tsx [hunk 1]
```diff
@@ -143,10 +143,10 @@ export const TokenInformationContextProvider = ({ children }: { children: React.
...
         tokenRegistryAddress,
         initialize,
-        holder,
-        beneficiary,
-        approvedBeneficiary,
-        approvedHolder,
+        holder: holder?.[0],
+        beneficiary: beneficiary?.[0],
+        approvedBeneficiary: approvedBeneficiary?.[0],
+        approvedHolder: approvedHolder?.[0],
```
Comment (justussoh) [inline src/common/contexts/TokenInformationContext/TokenInformationContext.tsx:168]: Not sure why but typescript says its this.

File: src/common/hooks/useEndorsementChain/useEndorsementChain.ts [hunk 2]
```diff
@@ -25,7 +25,7 @@ export const useEndorsementChain = (tokenRegistryAddress: string, tokenId: strin
       const logs = await provider.getLogs({ ...transferLogFilter, fromBlock: 0 });
       const parsedLogs = logs.map((log) => ({ ...log, ...tokenRegistry.interface.parseLog(log) }));
       const formattedLogs = parsedLogs.map((log) => {
-        const { blockNumber, values, transactionHash } = log;
+        const { blockNumber, args: values, transactionHash } = log;
```
Comment (justussoh) [inline src/common/hooks/useEndorsementChain/useEndorsementChain.ts:27]: values was rename to arg

File: src/services/etherjs/provider.ts [hunk 3]
```diff
@@ -2,15 +2,15 @@ import { ethers, providers } from "ethers";
 import { getLogger } from "../../utils/logger";
 const { trace } = getLogger("service:etherjs:provider");
 
-interface Ethereum extends providers.Web3Provider {
+interface Ethereum extends providers.ExternalProvider, providers.BaseProvider {
```
Comment (justussoh) [inline src/services/etherjs/provider.ts:5]: Web3Provider expected param of type `ExternalProvider` but windows.ethereum.on method was part of the BaseProvider class, not sure if this is right!

Comment (justussoh) [reply src/services/etherjs/provider.ts:5]: Previously, type  `provider.Web3Provider` looked more like the `windows.web3` structure, [refer here for more info](https://github.com/ethers-io/ethers.js/blob/master/packages/providers/lib/web3-provider.d.ts).

File: src/common/contexts/TokenInformationContext/TokenInformationContext.tsx (hunk 1 above)
Comment (Nebulis) [reply src/common/contexts/TokenInformationContext/TokenInformationContext.tsx:168]: I remind having this when I upgraded oa-verify to ethers v5 as weill. However when using token-registry or document-store package, no problem anymore :thinking:

**Review by rjchow — APPROVED** [review]

**Pull request merged**